miniml/compiler/compile.scm

```diff
@@ -1672,15 +1709,17 @@
 
 (define (compile-arg-default name default body)
   (let* ((noneline (cons (lid->pconstr "None" #nil) default))
-         (someline (cons (lid->pconstr "Some" (list name))
+         (someline (cons (lid->pconstr "Some" (list (lid->pvar name)))
                          (lid->evar name)))
          (default-expr (list 'EMatch (lid->evar name) (list noneline someline))))
     (list 'ELet #f (list (cons (lid->pvar name) default-expr)) body)))
 
 (define (compile-arg-pat pat name body)
-  (if (equal? (car pat) 'PVar)
-      body
-      (list 'EMatch (lid->evar name) (list (cons pat body)))))
+  (match pat
+   (('PVar _)
+    body)
+   (_
```

Comment:
What about the `'PWild` case?

Reply:
Currently, using `_` as a function argument will be transformed into a named argument, with `let _ = arg#n in ...` inserted in the function body. This is correct, slightly wasteful (but we quite probably don't care), and also a mild change of behavior compared to the pre-PWild-cleanup situation where `"_"` would happily be used as the argument name, even if used several times in a row. In the new code, at the low level I always use strings or #nil to represent local variables, and currently #nil is never used in function arguments. I would prefer to keep it that way, or at least to first do a check on the `compile-fundef(s)` code to ensure that it handles nil-arguments gracefully.

Reply:
Good enough for me, we can always change it again later if there are performance problems.